src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+        [](std::shared_ptr<LedgerKeyMeter> lkMeter1,
+           std::vector<LedgerEntry> entries,
+           UnorderedSet<LedgerKey> expectedSuccessKeys,
+           UnorderedSet<LedgerKey> expectedFailedKeys) {
+            // Some, but not all keys in the expectedSuccessKeys set should
+            // be able to load. We expect that the number of keys that can
+            // load is less than the size of the expectedFailedKeys set.
+            int successfulFailures = 0;
+            for (const auto& entry : entries)
+            {
+                auto k = LedgerEntryKey(entry);
+                auto size = xdr::xdr_size(entry);
+                if (expectedSuccessKeys.find(k) != expectedSuccessKeys.end())
+                {
+                    REQUIRE(lkMeter1->canLoad(k, size));
+                }
+                else if (lkMeter1->canLoad(k, size))
+                {
+                    successfulFailures++;
+                }
+            }
+            REQUIRE(successfulFailures <= expectedFailedKeys.size());
+        };
+
+    auto runTest = [&](size_t cacheSize) {
+        Config cfg = getTestConfig();
+        cfg.ENTRY_CACHE_SIZE = cacheSize;
+
+        // Test setup.
+        VirtualClock clock;
+        cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+        Application::pointer app = createTestApplication(clock, cfg);
+        UnorderedSet<LedgerKey> keysToPrefetch;
+        auto& root = app->getLedgerTxnRoot();
+        LedgerTxn ltx(root);
+
+        UnorderedSet<LedgerKey> expectedSuccessKeys;
+        UnorderedSet<LedgerKey> expectedFailedKeys;
+        std::set<LedgerEntry> entrySet;
+        std::set<LedgerKey> deadKeySet;
+
+        auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
+        auto lkMeterHot = std::make_shared<LedgerKeyMeter>();
+
+        auto entries =
+            LedgerTestUtils::generateValidUniqueLedgerEntriesWithExclusions(
+                {TTL}, 3);
+
+        auto deadKeys =
+            LedgerTestUtils::generateValidUniqueLedgerEntryKeysWithExclusions(
+                {CONFIG_SETTING}, 1);
+        auto ttlEntry = LedgerTestUtils::generateValidLedgerEntryOfType(TTL);
+        entries.push_back(ttlEntry);
+        for (auto e : entries)
+        {
+            e.lastModifiedLedgerSeq = 1;
+            ltx.createWithoutLoading(e);
+            entrySet.emplace(e);
+        };
+        for (const auto& k : deadKeys)
+        {
+            ltx.eraseWithoutLoading(k);
+            deadKeySet.emplace(k);
+        }
+
+        // Insert all entries into the database.
+        std::vector<LedgerEntry> ledgerVect{entrySet.begin(), entrySet.end()};
+        std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(),
+                                           deadKeySet.end()};
+
+        app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect,
+                                         deadKeyVect);
+        ltx.commit();
+
+        auto addTxn = [&](int32_t offset, std::vector<LedgerEntry> entries,
```

Comment:
Offset is always either `0` or `-1`, so why not just make this something like `bool enoughQuota`. You can avoid the overflow checks that way as well and should simplify things.